libs/common/ui/src/hooks/useConstObj.tsx

```diff
@@ -0,0 +1,18 @@
+import { useEffect, useState } from 'react'
+
+/**
+ * Will "retain" the reference to the same object,
+ * as long as the serialized representation remains the same.
+ * @param objInput Input to remain constant
+ * @returns a state reference to the `objInput`,
+ * static as long as the JSON stringlization of `objInput`
+ * remains the same(even if the reference is different).
+ */
+export function useConstObj<T extends object>(objInput: T) {
+  const [obj, setobj] = useState(objInput)
+  const objStr = JSON.stringify(obj)
```

Comment:
https://react.dev/reference/react/Component#shouldcomponentupdate-caveats
> We do not recommend doing deep equality checks or using JSON.stringify in shouldComponentUpdate. It makes performance unpredictable and dependent on the data structure of every prop and state. In the best case, you risk introducing multi-second stalls to your application, and in the worst case you risk crashing it.

Reply:
yeah, thats in the "should update" logic of class-based components, where there could be unserializable data passed to the component props, like react components and circular references. (which will break JSON.stringify). The object in this case is strongly typed, JSONable(since it come from database), and we are not overriding the existing prop checking within a component, we are just maintaining the `Object.is` reference. I don't think this quote is applicable in this context.

Reply:
My concern is deep equality is slower than shallow equality and this cost is paid on every render